server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -246,4 +268,18 @@ private void ensureDecommissionedAttributeHasZeroWeight(ClusterState state, Clus
             );
         }
     }
+
+    private void ensureNoVersionConflict(long requestedVersion, WeightedRoutingMetadata weightedRoutingMetadata) {
+        if ((weightedRoutingMetadata == null && requestedVersion != WeightedRoutingMetadata.INITIAL_VERSION)
+            || (weightedRoutingMetadata != null && requestedVersion != weightedRoutingMetadata.getVersion())) {
+            throw new UnsupportedWeightedRoutingStateException(
```

Comment:
Create a new exception for this maybe `VersionConflictWeightedRoutingException`

Reply:
I used the exception `UnsupportedWeightedRoutingStateException` and updated the details shown in the exception since its throwing 409. We can create a separate exception as well.